DiskSetup.sh

```diff
@@ -0,0 +1,341 @@
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-B' 'Boot-partition size (default: 500MiB)'
+      printf '\t%-4s%s\n' '-d' 'Base dev-node used for build-device'
+      printf '\t%-4s%s\n' '-f' 'Filesystem-type used for root filesystems (default: xfs)'
+      printf '\t%-4s%s\n' '-p' 'Comma-delimited string of colon-delimited partition-specs'
+      printf '\t%-6s%s\n' '' 'Default layout:'
+      printf '\t%-8s%s\n' '' '/:rootVol:4\n'
+      printf '\t%-8s%s\n' '' 'swap:swapVol:2'
+      printf '\t%-8s%s\n' '' '/home:homeVol:1'
+      printf '\t%-8s%s\n' '' '/var:varVol:2'
+      printf '\t%-8s%s\n' '' '/var/log:logVol:2'
+      printf '\t%-8s%s\n' '' '/var/log/audit:auditVol:100%%FREE'
+      printf '\t%-4s%s\n' '-r' 'Label to apply to root-partition if not using LVM (default: root_disk)'
+      printf '\t%-4s%s\n' '-v' 'Name assigned to root volume-group (default: VolGroup00)'
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--boot-size' 'See "-B" short-option'
+      printf '\t%-20s%s\n' '--disk' 'See "-d" short-option'
+      printf '\t%-20s%s\n' '--fstype' 'See "-f" short-option'
+      printf '\t%-20s%s\n' '--partition-string' 'See "-p" short-option'
+      printf '\t%-20s%s\n' '--rootlabel' 'See "-r" short-option'
+      printf '\t%-20s%s\n' '--vgname' 'See "-v" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+# Partition as LVM
+function CarveLVM {
+   local ITER
+   local MOUNTPT
+   local PARTITIONARRAY
+   local PARTITIONSTR
+   local VOLFLAG
+   local VOLNAME
+   local VOLSIZE
+
+   # Whether to use flag-passed partition-string or default values
+   if [ -z ${GEOMETRYSTRING+x} ]
+   then
+       # This is fugly but might(??) be easier for others to follow/update
+       PARTITIONSTR="/:rootVol:4"
+       PARTITIONSTR+=",swap:swapVol:2"
+       PARTITIONSTR+=",/home:homeVol:1"
+       PARTITIONSTR+=",/var:varVol:2"
+       PARTITIONSTR+=",/var/log:logVol:2"
+       PARTITIONSTR+=",/var/log/audit:auditVol:100%FREE"
+   else
+       PARTITIONSTR="${GEOMETRYSTRING}"
+   fi
+
+   # Convert ${PARTITIONSTR} to iterable array
+   IFS=',' read -r -a PARTITIONARRAY <<< "${PARTITIONSTR}"
+
+   # Clear the MBR and partition table
+   err_exit "Clearing existing partition-tables..." NONE
+   dd if=/dev/zero of="${CHROOTDEV}" bs=512 count=1000 > /dev/null 2>&1 || \
+     err_exit "Failed clearing existing partition-tables"
+
+   # Lay down the base partitions
+   err_exit "Laying down new partition-table..." NONE
+   parted -s "${CHROOTDEV}" -- mktable gpt \
+      mkpart primary "${FSTYPE}" 2048s "${BOOTDEVSZ}" \
+      mkpart primary "${FSTYPE}" "${BOOTDEVSZ}" 100% \
+      set 1 bios_grub on \
+      set 2 lvm || \
+     err_exit "Failed laying down new partition-table"
+
+   ## Create LVM objects
+
+   # Let's only attempt this if we're a secondary EBS
+   if [[ ${CHROOTDEV} == /dev/xvda ]] || [[ ${CHROOTDEV} == /dev/nvme0n1 ]]
```

Comment:
curious to see how this will work in spel, which is intentionally rewriting the primary EBS...

Reply:
Had it there to prevent scribbling the root-drive during interactive use. Guess I'll further wrap it in a "is this an interactive invocation" block when the time comes.